### DISCUSSION TITLE
No config file is .ssh folder

### DISCUSSION THREAD

**Discussion by ok-cool** [community/community#22589] [New to GitHub]
<p>Following this GitHub <a href="https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/" rel="nofollow noopener">guide</a>, I’m stucked at the “Adding your SSH key to the ssh-agent” section, step 2: I just don’t have config file in my .ssh folder. Should I create .txt file named “config” and paste the following lines into it? I’m using macOS Sierra 10.14.1.</p>

**Comment by CleanCut** [comment, marked as answer]
<p>Yes!  That is exactly what you should do. Create a file named config in your .ssh folder using your text editor of choice and paste in the lines from the guide and then save the file.</p>

**Reply by Mansoor-Saeed** [reply]
Thanks a lot

**Reply by kabeerasim** [reply]
Didn't work even after creating a config.txt file. When I try to ssh, it still says "no matching host key type found. Their offer: ssh-rsa,ssh-dss" After looking on a forum I came across this thing where I should add:

Host nas
  HostName 192.168.239.133
  HostKeyAlgorithms=+ssh-dss
  
 This to my config file. Any Idea what to do?

**Reply by CleanCut** [reply]
The file should literally be named `config` (no `.txt` extension) and be stored in the `.ssh/` folder.

**Reply by kabeerasim** [reply]
At first I do created the file without the extension, but still the ssh wasn't ready configurations fr here, so I added the extension. But still, nothing worked out. Getting the same "no matching host key found, their offer rsa-dss etc..." error even after adding:
Host [IP_ADDR]
HostKeyAlgorithms=+ssh-dss
To the config file.

**Reply by tomeewwolffart** [reply]
For me it worked now - the filename is only config with no extension and put under userprofile /.ssh for the current user to work. SSH picked some parameters up successfully.

**Comment by AndreiCherniaev** [comment]
I add pubkey to github via [FireFox](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account), after
```
ssh-add /home/username/.ssh/mygit/mygithub
git config --global user.signingkey /home/username/.ssh/mygit/mygithub.pub
git config --global user.signingkey /home/username/.ssh/mygit/mygithub.pub
```
And restart...

now I can clone like
`git clone git@github.com:libconfuse/libconfuse.git`

**Comment by akashdip2001** [comment]
Yes, you can create a `config` file in your `.ssh` folder if it doesn't exist. On macOS, the `.ssh` folder is usually located in your home directory (`~`). 

Here are the steps you can follow:

1. Open your terminal.

2. Navigate to your `.ssh` directory:
   ```bash
   cd ~/.ssh
   ```

3. Create the `config` file:
   ```bash
   touch config
   ```

4. Open the `config` file in a text editor of your choice (you can use `nano`, `vim`, or any text editor you prefer):
   ```bash
   nano config
   ```

5. Paste the following lines into the `config` file:
   ```plaintext
   Host *
     AddKeysToAgent yes
     UseKeychain yes
     IdentityFile ~/.ssh/id_rsa
   ```
   This configuration is telling SSH to automatically add your private key (`id_rsa`) to the SSH agent and use the macOS keychain.

6. Save and exit the text editor.

Now you've created a `config` file with the necessary settings. You can proceed with the other steps in the guide to add your SSH key to the ssh-agent.

Remember, the `config` file is not a text file with a `.txt` extension; it's a configuration file without an extension. Make sure to save it without any file extension.

## Profile
![Profile Image](https://thumbnails.odycdn.com/optimize/s:200:0/quality:95/plain/https://thumbs.odycdn.com/bf58cf4d7ec52c62f02c52e16f150a5b.webp)
**AKASHDIP MAHAPATRA**
"everything is physics bro"  
E=mc²

|[Subscribe on YouTube](https://www.youtube.com/channel/UCxvmp634YDc41xCWOdvWqoQ?sub_confirmation=1) |
|----  |
